### ISSUE TITLE
Incorrect name matching

### ISSUE THREAD

**Issue by nickdos** [AtlasOfLivingAustralia/specieslist-webapp#59]
When trying to add a species "Corvus orru" to a list, it creates an entry for "Corvus orru orru".

Would be good to be able to pass in a rank to assist with the match.

Related to #3.

**Comment by m-hope** [comment]
There are a number of examples of names within a list not being recognised even though they are a valid name and recognised in the ALA search bar.  Examples include:

- Scoparia dulcis in https://lists.ala.org.au/speciesListItem/list/dr7802
- a large number (164) of genera in https://lists.ala.org.au/speciesListItem/list/dr7709?fq=guid%3Anull

**Comment by m-hope** [comment]
As the Lists tool is now being used by Profiles (especially the eFlora of Australia), this issue needs to be investigated more urgently.

**Comment by charvolant** [comment]
Yes, it looks like the parent-child synonym issue. Corvus orru orru has a synonym of Corvus orru because someone has decided that what was originally described as a species is now a subspecies. The ala-name-matching algorithms flag this with a parent-child-synonym exception/indicator in the biocache but its not very helpful when list matching.

Possible solution is to bypass parent-child synonym unpleasantness altogether?

**Comment by nickdos** [comment]
@charvolant - There are also a bunch of genus names which are not recognised - could you check those as well please (second comment above)?

**Comment by charvolant** [comment]
I've tried a test:

Acaena creates a homonym exception. We have

* https://bie.ala.org.au/species/http://id.biodiversity.org.au/node/apni/7834306 (what we want)
* https://bie.ala.org.au/species/http://id.biodiversity.org.au/name/apni/71997 (a section recorded in APNI)
* https://bie.ala.org.au/species/http://id.biodiversity.org.au/name/apni/71999 (a subsection in APNI)

More to follow
